javaparser-core/src/main/javacc/java.jj

```diff
@@ -1318,15 +1329,30 @@ CompilationUnit CompilationUnit():
 	NodeList<ImportDeclaration> imports = emptyList();
 	ImportDeclaration in = null;
 	NodeList<TypeDeclaration<?>> types = emptyList();
+   ModifierHolder modifier;
 	TypeDeclaration tn = null;
+	ModuleDeclaration module = null;
 }
 {
     ( ";" )*
     [ LOOKAHEAD(PackageDeclaration()) pakage = PackageDeclaration() ]
     ( in = ImportDeclaration() { imports = add(imports, in); } | ";" )*
-    ( tn = TypeDeclaration() { types = add(types, tn); } | ";")*
+    (
+        modifier = Modifiers()
+        (
+            tn = ClassOrInterfaceDeclaration(modifier) { types = add(types, tn); }
+        |
+            tn = EnumDeclaration(modifier) { types = add(types, tn); }
+        |
+            tn = AnnotationTypeDeclaration(modifier) { types = add(types, tn); }
+        |
+            module = ModuleDeclaration(modifier)
+        |
```

Comment:
This is what I did to make CompilationUnit() match all possible content. So the grammar thinks it's okay to mix classes, multiple module declarations, enums and annotation declarations in the same file. Yet again, a validator could check if a module declaration is mixed with type declarations and flag a warning or error.